View/ViewHandler.php

```diff
@@ -132,6 +133,15 @@ public function __construct(
         $defaultEngine = 'twig',
         array $options = []
     ) {
+        if (!is_null($templating) && !$templating instanceof EngineInterface && !$templating instanceof Environment) {
+            throw new \InvalidArgumentException(sprintf(
```

Comment:
Let's throw a `TypeError` here instead. That will be consistent with how it's done in the Symfony code.